pkg/apis/autoscaling/v1alpha1/pa_lifecycle_test.go

```diff
@@ -977,16 +977,27 @@ func TestTypicalFlow(t *testing.T) {
 	r.InitializeConditions()
 	apistest.CheckConditionOngoing(r, PodAutoscalerConditionActive, t)
 	apistest.CheckConditionOngoing(r, PodAutoscalerConditionReady, t)
+	// Initializing the condition set shouldn't have an effect on HasBeenActive
```

Comment:
@vagababov are these sufficient to make sure changing the PA activeness won't affect HasBeenActive?

Reply:
We need to check value, rather than just nil/not-nil

Reply:
Since PodAutoscalerConditionHasBeenActive isn't part of the PA condition set initialization, it will be nil to begin with. We are checking for its value down below after marking HasBeenActive (https://github.com/knative/serving/pull/8403/files/62fe51705659aad952ec2293fbd98a5fbebe98f1#diff-7b6c8b6b6a368eb44168d5d5ac3da3edR994), but here since it's uninitialized we can only verify that it is nil.